Excel_Adapter/AdapterActions/Pull.cs

```diff
@@ -36,13 +37,10 @@ public partial class ExcelAdapter : BHoMAdapter
         /**** Method Overrides                          ****/
         /***************************************************/
 
-        public override IEnumerable<object> Pull(IRequest request, PullType pullOption = PullType.AdapterDefault, ActionConfig actionConfig = null)
+        public override IEnumerable<object> Pull(IRequest request = null, PullType pullOption = PullType.AdapterDefault, ActionConfig actionConfig = null)
         {
-            if (request == null)
-            {
-                BH.Engine.Base.Compute.RecordError("The request has not been provided for Pull action.");
-                return new List<object>();
-            }
+            if (request == null || request is FilterRequest)
+                request = new CellValuesRequest();
```

Comment:
I assume the check for FIlterRequest here is added as null is never provided (always a fitlerrequest) and this aims to make it so that nothing provided -> pull first sheet in file, which I agree with. 
However, providing a FilterRequest or a Type to the pull simple follows this, without any warning/error/similar.

Think this can simply be fixed by one of the two suggestions:
```suggestion
            if (request == null)
                request = new CellValuesRequest();
            else if (request is FilterRequest)
            {
                FilterRequest filterReq = request as FilterRequest;
                if (filterReq.Type == null)
                    request = new CellValuesRequest();
                else
                    request = new ObjectRequest { ObjectType = filterReq.Type };
            }

```

or

```suggestion
            if (request == null)
                request = new CellValuesRequest();
            else if (request is FilterRequest)
            {
                FilterRequest filterReq = request as FilterRequest;
                if (filterReq.Type == null)
                    request = new CellValuesRequest();
                else
                {
                    BH.Engine.Base.Compute.RecordError("FilterRequests are not supported by the ExcelAdapter.");
                    return new List<object>();
                }
            }
```

Both above should keep the principle of pulling everything if a FilterRequest with a null type is provided, the first one converting it to an object request if the type is provided, and the second one providing an error if the type is provided (not empty FilterRequest ). 

Happy for this to be dealt with in some other way as well.

Potential better way to deal with this is to instead override this method in the adapter:

https://github.com/BHoM/BHoM_Adapter/blob/177ee3915a21d702aa76e3dd791d7b1dceffe8e4/BHoM_Adapter/AdapterActions/Pull.cs#L46-L61

To give the wanted behaviour.

Reply:
Thanks @IsakNaslundBh , the second option is great. Will commit.